src/xlsx.js

```diff
@@ -0,0 +1,130 @@
+export class Workbook {
+  constructor(workbook) {
+    Object.defineProperties(this, {
+      _: {value: workbook},
+      sheetNames: {
+        value: workbook.worksheets.map((sheet) => sheet.name),
+        writable: false,
+        enumerable: true,
+      },
+    });
+  }
+  sheet(name, {range, headers = false} = {}) {
+    const sname =
+      typeof name === "number"
+        ? this.sheetNames[name]
+        : this.sheetNames.includes(name + "")
+        ? name + ""
+        : null;
+    if (sname == null) throw new Error(`Sheet not found: ${name}`);
+    const sheet = this._.getWorksheet(sname);
+    return extract(sheet, {range, headers});
+  }
+}
+
+function extract(sheet, {range, headers}) {
+  let [[c0, r0], [c1, r1]] = parseRange(range, sheet);
+  const headerRow = headers && sheet._rows[r0++];
+  let names = new Set(["#"]);
+  for (let n = c0; n <= c1; n++) {
+    let name = (headerRow ? valueOf(headerRow._cells[n]) : null) || AA(n);
+    while (names.has(name)) name += "_";
+    names.add(name);
+  }
+  names = new Array(c0).concat(Array.from(names));
+
+  const output = new Array(r1 - r0 + 1);
+  for (let r = r0; r <= r1; r++) {
+    const row = (output[r - r0] = Object.defineProperty({}, "#", {
+      value: r + 1,
+    }));
+    const _row = sheet._rows[r];
+    if (_row && _row.hasValues)
+      for (let c = c0; c <= c1; c++) {
+        const value = valueOf(_row._cells[c]);
+        if (value != null) row[names[c + 1]] = value;
+      }
+  }
+
+  output.columns = names.filter(() => true); // Filter sparse columns
+  return output;
+}
+
+function valueOf(cell) {
+  if (!cell) return;
+  const {value} = cell;
+  if (value && value instanceof Date) return value;
+  if (value && typeof value === "object") {
+    if (value.formula || value.sharedFormula)
+      return value.result && value.result.error ? NaN : value.result;
+    if (value.richText) return value.richText.map((d) => d.text).join("");
+    if (value.text)
+      return value.hyperlink
+        ? `<a href="${encodeURI(value.hyperlink)}">${escapeHTML(
+            value.text
+          )}</a>`
```

Comment:
We’re losing type information here: the caller won’t be able to distinguish between plain text and HTML. This gets to my “big question” at the top of the review. Some possible alternatives: return an HTML element;  ignore the hyperlink and return the text; return the object returned by ExcelJS rather than a simplified representation.

Reply:
I know of two common cases for hyperlinks in Excel:

1. As documentation to link to more information
2. As data in a row

Simplifying this to return a simple string and also not to duplicate information when the hyperlink and text are identical. For the "data in a row" case, this feels the least obtrusive and most natural.

But your original point still stands: we're still losing that the text explicitly contains a hyperlink and the user will need to re-infer it's a URL.